mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/impl/BlobMappedGridTypeDescriptor.java

```diff
@@ -39,22 +42,15 @@ protected void doBind(Tuple resultset, X value, String[] names, WrapperOptions o
 	@Override
 	public <X> GridValueExtractor<X> getExtractor(final JavaTypeDescriptor<X> javaTypeDescriptor) {
 		return new GridValueExtractor<X>() {
-
 			@Override
-			public X extract(Tuple tuple, String name, WrapperOptions options) {
-				final ByteArrayOutputStream binaryStream = (ByteArrayOutputStream) tuple.get( name );
-				if ( binaryStream == null ) {
-					return null;
-				}
-				else {
-					byte[] data = binaryStream.toByteArray();
-					return javaTypeDescriptor.wrap( data, options );
+			public X extract(Tuple tuple, String name) {
+				final GridFSDownloadStream binaryStream = (GridFSDownloadStream) tuple.get( name );
+				if ( binaryStream != null ) {
+					long fileSize  = binaryStream.getGridFSFile().getLength();
+					Blob blob = BlobProxy.generateProxy( binaryStream, fileSize );
```

Comment:
I don't think you  need or are supposed to use the generateProxy method directly. You can probably rewrite this in the following way:
```
	@Override
	public <X> GridValueExtractor<X> getExtractor(final JavaTypeDescriptor<X> javaTypeDescriptor) {
		return new GridValueExtractor<X>() {

			@Override
			public X extract(Tuple tuple, String name, WrapperOptions options) {
				final InputStream binaryStream = (InputStream) tuple.get( name );
				if ( binaryStream == null ) {
					return null;
				}
				else {
					return javaTypeDescriptor.wrap( binaryStream, options );
				}
			}

			@Override
			public X extract(Tuple resultset, String name) {
				throw new UnsupportedOperationException( "Resultset cannot be converted without a LobCreator object" );
			}
		};
	}
```
Because this is a Blob type the proxy is generated when the wrap is called and it uses the right options to generate it.
This class should probabaly be rename as `BinaryStreamMappedGridTypeDescriptor` and I suspect you can reuse it for the `String` and `byte[]` cases. Not 100% sure though